src/go/pkg/robotauth/robotauth.go

```diff
@@ -150,3 +169,40 @@ func (r *RobotAuth) CreateRobotTokenSource(ctx context.Context) oauth2.TokenSour
 	}
 	return c.TokenSource(ctx)
 }
+
+// CreateJWT allows to create a JWT for authentication against the token vendor.
+// This does not grant Google Cloud access, but can be used for for explicit
+// authentication with the token vendor.
+func (r *RobotAuth) CreateJWT(ctx context.Context, lifetime time.Duration) (string, error) {
+	p, _ := pem.Decode(r.PrivateKey)
+	if p == nil {
+		return "", fmt.Errorf("decode private key")
+	}
+	parsedKey, err := x509.ParsePKCS8PrivateKey(p.Bytes)
+	if err != nil {
+		parsedKey, err = x509.ParsePKCS1PrivateKey(p.Bytes)
+		if err != nil {
+			return "", fmt.Errorf("private key should be a PEM or plain PKCS1 or PKCS8; parse error: %v", err)
+		}
+	}
+	parsed, ok := parsedKey.(*rsa.PrivateKey)
+	if !ok {
+		return "", fmt.Errorf("private key is invalid")
+	}
+
+	claimSet := &jws.ClaimSet{
+		Iss:   r.PublicKeyRegistryId,
+		Scope: r.getTokenEndpoint(),
```

Comment:
This is /apis/core.token-vendor/v1/token.oauth2, should it be jwt.verify instead? I don't know what the exact intent of scope/aud is, I'm just pattern matching on the strings.

Reply:
While I think this would be technically better, I'm shying away from having a second audience on the token-vendor side, since it's currently provided as cmdline arg.

There's no access ability difference on the metadata-server, so I don't think we have a real gain from splitting it.
Though I can still do it for cleanliness.

Reply:
Thanks, please leave a comment then to explain that it's an intentional simplification as it doesn't affect the use.